src/ring_theory/simple_module/finite_dimensional.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import ring_theory.simple_module.basic
+import linear_algebra.finite_dimensional
+import algebra.module.submodule.lattice
+
+/-!
+# 1-d modules are simple
+
+We prove this for modules over a `k`-algebra, when `k` is a field,
+and separately for modules over a division ring.
+
+(Finding the common generalisation would be welcome.)
+-/
+
+open finite_dimensional
+
+/-- Any `k`-algebra module with positive dimension over `k`
+has a nontrivial submodule lattice, since `⊥ ≠ ⊤`. -/
+lemma submodule_nontrivial_of_finrank_pos {k : Type*} [field k] {A : Type*} [ring A] [algebra k A]
+  {M : Type*} [add_comm_group M] [module k M] [module A M] [is_scalar_tower k A M]
+  (h : 0 < finrank k M) : nontrivial (submodule A M) :=
+⟨⟨⊥, ⊤, begin
+      apply_fun (λ m : submodule A M, finrank k (submodule.restrict_scalars k m)),
+      dsimp,
+      simp [h.ne],
+    end⟩⟩
+
+/-- Any module over a division ring with positive dimension over `k`
+has a nontrivial submodule lattice, since `⊥ ≠ ⊤`. -/
+lemma submodule_nontrivial_of_finrank_pos' {k : Type*} [division_ring k]
+  {M : Type*} [add_comm_group M] [module k M]
+  (h : 0 < finrank k M) : nontrivial (submodule k M) :=
+⟨⟨⊥, ⊤, begin
+      apply_fun (λ m : submodule k M, finrank k m),
+      dsimp,
+      simp [h.ne],
+    end⟩⟩
+
+example (a : ℕ) (h : a = 1) : 0 < a := nat.lt_of_sub_eq_succ h
+
+/-- Any `k`-algebra module which is 1-dimensional over `k` is simple. -/
+lemma is_simple_module_of_finrank_eq_one {k : Type*} [field k] {A : Type*} [ring A] [algebra k A]
+  {M : Type*} [add_comm_group M] [module k M] [module A M] [is_scalar_tower k A M]
+  (h : finrank k M = 1) : is_simple_module A M :=
```

Comment:
I [added a more general version to *linear_algebra/finite_dimensional* in #17401](https://github.com/leanprover-community/mathlib/pull/17401/commits/d525d847669c5150cb55de8ba8a2ea625e9deaa1#diff-a460505ea5f3db30574d1f6c292ad14ceffb53a09ba980303313df40128bc521R1620-R1621) assuming only `[division_ring k] [semiring A] [has_smul k A] [is_scalar_tower k A M]`, so that it automatically applies in the situation of the next lemma. My version is stated with `is_simple_order` because `is_simple_module` isn't imported there, which should be okay since both are reducibly defeq.

Reply:
Okay! Looking forward to #17401, and then I'll see if there's anything to salvage here!

Reply:
I'll probably wait for the refactors #17481, #17482 (and possibly #17476) to be merged before merging master into #17401.